api/api_test.go

```diff
@@ -29,3 +37,94 @@ func testHTTPServer(
 
 	return config, ln
 }
+
+// nextPort is the next port to use for the API server.
+var nextPort int32 = 28200
+
+// restVaultServer runs an instance of the Vault server in development mode.
+// This requires that the vault binary is installed and in the $PATH.
+func testVaultServer(t *testing.T) (*Client, func()) {
+	bin, err := exec.LookPath("vault")
+	if err != nil || bin == "" {
+		t.Fatal("vault binary not found")
+	}
+
+	// Get the port number
+	port := atomic.AddInt32(&nextPort, 1)
+
+	// Construct the address
+	addr := fmt.Sprintf("127.0.0.1:%d", port)
+
+	// Start the server
+	cmd := exec.Command(
+		bin, "server", "-dev",
+		"-dev-listen-address", addr,
+		"-dev-root-token-id", "root",
+	)
+	if err := cmd.Start(); err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	for i := 0; i < 10; i++ {
+		conn, err := net.DialTimeout("tcp", addr, time.Second)
+		if err != nil {
+			time.Sleep(100 * time.Millisecond)
+			continue
+		}
+		conn.Close()
+
+		config := DefaultConfig()
+		config.Address = fmt.Sprintf("http://%s", addr)
+		client, err := NewClient(config)
+		if err != nil {
+			t.Fatal(err)
+		}
+		client.SetToken("root")
+
+		return client, func() {
+			cmd.Process.Signal(os.Interrupt)
+			cmd.Process.Wait()
+		}
+	}
+
+	t.Fatalf("timeout waiting for vault server")
+	return nil, nil
+}
+
+func testPostgresDatabase(t *testing.T) (string, func()) {
```

Comment:
We already have another place where we invoke a postgres database using dockertest. Rather than put this in the API package it probably would be good to pick one of the two implementations and put it in a helper package that can be called by both.

Using a `helpers/testing` package to hold both testing common code and as a launchpoint for tests that can't easily be run in-package (e.g. ones using `vault.TestCluster`) might be the right approach.